### ISSUE TITLE
Devdocs: READMEs from `calypso-ui` not shown

### ISSUE THREAD

**Issue by ockham** [Automattic/wp-calypso#35809]
Found by @sirreal at https://github.com/Automattic/wp-calypso/pull/35768#pullrequestreview-279575337.

#### Steps to reproduce
Go to https://wpcalypso.wordpress.com/devdocs/design/progress-bar

#### What I expected

The component's README to be loaded

#### What happened instead

It says 'No documentation available.' at the bottom of the page, even though there is https://github.com/Automattic/wp-calypso/blob/053e45ed6d06a27f258e6254025c6584550d8fc4/packages/calypso-ui/src/progress-bar/README.md

#### Browser / OS version

(All)

#### Screenshot / Video

![image](https://user-images.githubusercontent.com/96308/63760519-49a11100-c8e1-11e9-8558-5d42bb5751d2.png)

#### Context / Source

This isn't quite as straight-forward to fix as it seems :sweat_smile: 

Relevant code: https://github.com/Automattic/wp-calypso/blob/053e45ed6d06a27f258e6254025c6584550d8fc4/client/devdocs/design/search-collection.jsx#L39-L46

https://github.com/Automattic/wp-calypso/blob/053e45ed6d06a27f258e6254025c6584550d8fc4/client/devdocs/design/index.jsx#L255

I was thinking of something like 

```diff
diff --git a/client/devdocs/design/index.jsx b/client/devdocs/design/index.jsx
index e9017cce34..1ef1650067 100644
--- a/client/devdocs/design/index.jsx
+++ b/client/devdocs/design/index.jsx
@@ -252,7 +252,7 @@ class DesignAssets extends React.Component {
 					<PlansSkipButton readmeFilePath="plans/plans-skip-button" />
 					<PodcastIndicator readmeFilePath="podcast-indicator" />
 					<Popovers readmeFilePath="popover" />
-					<ProgressBar readmeFilePath="progress-bar" />
+					<ProgressBar readmeFilePath="@automattic/calypso-ui" />
 					<PromoSection readmeFilePath="promo-section" />
 					<PromoCard readmeFilePath="promo-section/promo-card" />
 					<Ranges readmeFilePath="forms/range" />
diff --git a/client/devdocs/design/search-collection.jsx b/client/devdocs/design/search-collection.jsx
index 632ce54444..e5563b05a5 100644
--- a/client/devdocs/design/search-collection.jsx
+++ b/client/devdocs/design/search-collection.jsx
@@ -37,11 +37,18 @@ const shouldShowInstance = ( example, filter, component ) => {
 };
 
 const getReadmeFilePath = ( section, example ) => {
+	const { readmeFilePath } = example.props;
+
 	switch ( section ) {
 		case 'design':
-			return `/client/components/${ example.props.readmeFilePath }/README.md`;
+			if ( readmeFilePath && readmeFilePath.startsWith( '@automattic/' ) ) {
+				// Component is in a package: treat the path as absolute.
+				console.log( readmeFilePath + 'README.md', require.resolve( readmeFilePath ) );
+				return readmeFilePath + '/README.md';
+			}
+			return `/client/components/${ readmeFilePath }/README.md`;
 		default:
-			return `/client/${ section }/${ example.props.readmeFilePath }/README.md`;
+			return `/client/${ section }/${ readmeFilePath }/README.md`;
 	}
 };
 
```

However, module resolution has trouble with the `@automattic/calypso-ui` path here (and I didn't really want to hardcode it to `packages/calypso-ui`).

I'm now thinking that maybe our best option is to replace the `devdocs/service/content` endpoint that is used to obtain those READMEs (and that's one of the oldest parts of Calypso, long predating Webpack IIRC) by a Webpack loader

https://github.com/Automattic/wp-calypso/blob/053e45ed6d06a27f258e6254025c6584550d8fc4/server/devdocs/index.js#L212-L243

**Comment by sirreal** [comment]
Is this something where we could plug in `require.resolve` and be done?

**Comment by ockham** [comment]
Closed by #38239.
